### PR TITLE
e2e update with contactCallResults

### PR DESCRIPTION
**Pull request by nikolovyanko**
E2E additional contactResult check for child records.
[003239](https://github.com/hashgraph/hedera-services/issues/3239)

### CODE REVIEW

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ContractBurnHTSSuite.java [hunk 1]
```diff
@@ -359,10 +408,26 @@ private HapiApiSpec HSCS_PREC_020_rollback_burn_that_fails_after_a_precompile_tr
...
+																.withStatus(SUCCESS)
+																.withTotalSupply(1)
+														)
+										),
 								recordWith()
 										.status(INSUFFICIENT_SENDER_ACCOUNT_BALANCE_FOR_CUSTOM_FEE)
+										.contractCallResult(
+												resultWith()
+														.contractCallResult(htsPrecompileResult()
+//																.forFunction(HTSPrecompileResult.FunctionType.FAILED)
```
Comment (IvanKavaldzhiev) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ContractBurnHTSSuite.java:427]: Probably we don't need this comment?

**Review by IvanKavaldzhiev — DISMISSED** [review]
LGTM!

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ContractBurnHTSSuite.java (hunk 1 above)
Comment (nikolovyanko) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ContractBurnHTSSuite.java:427]: Yes, I wanted to ask about the commented row : is "HTSPrecompileResult.FunctionType.FAILED" with adequate value?

**Review by mustafauzunn — APPROVED** [review]
LGTM

**Review by shemnon — APPROVED** [review]

**Pull request merged**